Excel-custom-functions/Batching/samplewithtodos.js

```diff
@@ -0,0 +1,76 @@
+//Custom Functions Batching Sample with TODO items
+
+//TODO #1: Create custom functions for adding and multiplying numbers.
+
+function _pushOperation(op, args) {
+    var invocationEntry = {
+      "operation": op, // e.g. sum
+      "arguments": args,
+      "resolve": undefined,
+      "reject": undefined
+    };
+  
+    // Each invocation has a unique promise, which is mapped to the resolve and reject
+    // properties in the invocationEntry object. 
+    var promise = new Promise((resolve, reject) => { 
+      invocationEntry.resolve = resolve; 
+      invocationEntry.reject = reject;
+    });
+  
+  
+    _batch.push(invocationEntry);
+  
+    // TODO #2: If a remote request hasn't been scheduled yet,
+    // schedule it after a certain timeout, e.g. 2 sec.
+    
+    //Return the promise for this invocation.
+    return promise;
+  }
+  
+  var _batch = [];
+  var _isBatchedRequestScheduled = false;
+  
+  // TODO #3: Write a function which creates a copy of the batch array, empty the batch array,
```

Comment:
Need verb tenses to match: creates, empties, creates. or create, empty, create, pick one or the other.